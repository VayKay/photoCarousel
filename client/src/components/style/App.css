.photoCarousel {
    font-family: Circular, -apple-system, BlinkMacSystemFont, Roboto, Helvetica Neue, sans-serif;
    font-size: 14px;
    line-height: 1.43;
    color: #484848;
    background-color: #fff;
    margin-top: -10;
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
}

.topButtonsContainer {
    position: absolute !important;
    top: 24px !important;
    right: 24px !important;
    z-index: 1 !important;
}

.topButtonsInner {
    white-space: nowrap !important;
    line-height: 0 !important;
}

.buttonOneContainer {
    display: inline-block !important;
}

.buttonOneInner {
    display: inline-block !important;
    box-shadow: 0 1px 1px 1px rgba(0, 0, 0, 0.14) !important;
    border-radius: 4px !important;
}

.buttonOne {
    border-radius: var(--border-button-small-border-radius, 4px) !important;
    font-size: var(--font-button-small-font-size, 14px) !important;
    line-height: var(--font-button-small-line-height, 22px) !important;
    letter-spacing: var(--font-button-small-letter-spacing, normal) !important;
    font-family: var(--font-button-small-font-family, Circular,-apple-system,BlinkMacSystemFont,Roboto,Helvetica Neue,sans-serif) !important;
    text-transform: var(--font-button-small-text-transform, undefined) !important;
    padding-top: var(--spacing-button-small-vertical, 6px) !important;
    padding-bottom: var(--spacing-button-small-vertical, 6px) !important;
    font-weight: var(--font-book-font-weight, 600) !important;
    border-width: var(--border-button-small-border-width, 1px) !important;
    padding-left: var(--spacing-button-small-horizontal, 15px) !important;
    padding-right: var(--spacing-button-small-horizontal, 15px) !important;
    box-shadow: var(--shadow-button-level0-box-shadow, none) !important;
    background: var(--color-buttons-inverse-color, #ffffff) !important;
    border-color: var(--color-buttons-inverse-border, transparent) !important;
    color: var(--color-buttons-inverse-text, #484848) !important;
}

.buttonOne {
    display: inline-block !important;
    margin: 0px !important;
    position: relative !important;
    text-align: center !important;
    text-decoration: none !important;
    -webkit-transition-property: background, border-color, color !important;
    -moz-transition-property: background, border-color, color !important;
    transition-property: background, border-color, color !important;
    -webkit-transition-duration: 0.2s !important;
    transition-duration: 0.2s !important;
    -webkit-transition-timing-function: ease-out !important;
    transition-timing-function: ease-out !important;
    width: auto !important;
    cursor: pointer !important;
    border-radius: 4px !important;
    font-size: 14px !important;
    line-height: 22px !important;
    letter-spacing: normal !important;
    font-family: Circular,-apple-system,BlinkMacSystemFont,Roboto,Helvetica Neue,sans-serif !important;
    text-transform: undefined !important;
    padding-top: 6px !important;
    padding-bottom: 6px !important;
    font-weight: 600 !important;
    border-width: 1px !important;
    border-style: solid !important;
    padding-left: 15px !important;
    padding-right: 15px !important;
    min-width: 48.54101966249685px !important;
    box-shadow: none !important;
    background: #ffffff !important;
    border-color: transparent !important;
    color: #484848 !important;
}

.buttonOneSpan {
    font-size: var(--font-button-small-font-size, 14px) !important;
    line-height: var(--font-button-small-line-height, 22px) !important;
    letter-spacing: var(--font-button-small-letter-spacing, normal) !important;
    font-family: var(--font-button-small-font-family, Circular,-apple-system,BlinkMacSystemFont,Roboto,Helvetica Neue,sans-serif) !important;
    text-transform: var(--font-button-small-text-transform, undefined) !important;
    padding-top: var(--font-button-small-padding-top, undefined) !important;
    padding-bottom: var(--font-button-small-padding-bottom, undefined) !important;
    font-size: 14px !important;
    line-height: 22px !important;
    letter-spacing: normal !important;
    font-family: Circular,-apple-system,BlinkMacSystemFont,Roboto,Helvetica Neue,sans-serif !important;
    text-transform: undefined !important;
    color: inherit !important;
    padding-top: undefined !important;
    padding-bottom: undefined !important;
}

.buttonOneText {
    display: -webkit-box !important;
    display: -moz-box !important;
    display: -ms-flexbox !important;
    display: -webkit-flex !important;
    display: flex !important;
}

.buttonOneIcon {
    -ms-flex-item-align: center !important;
    display: -webkit-inline-box !important;
    display: -moz-inline-box !important;
    display: -ms-inline-flexbox !important;
    display: -webkit-inline-flex !important;
    display: inline-flex !important;
    -webkit-align-self: center !important;
    align-self: center !important;
}

.buttonOneShare {
    -ms-flex-item-align: center !important;
    display: -webkit-inline-box !important;
    display: -moz-inline-box !important;
    display: -ms-inline-flexbox !important;
    display: -webkit-inline-flex !important;
    display: inline-flex !important;
    -webkit-align-self: center !important;
    align-self: center !important;
}

.buttonTwoContainer {
    display: inline-block !important;
}

.buttonTwoInner {
    display: inline-block !important;
    box-shadow: 0 1px 1px 1px rgba(0, 0, 0, 0.14) !important;
    border-radius: 4px !important;
}

.buttonTwoAccessSpan {
    border: 0px !important;
    clip: rect(0 0 0 0) !important;
    -webkit-clip-path: inset(100%) !important;
    clip-path: inset(100%) !important;
    height: 1px !important;
    overflow: hidden !important;
    padding: 0px !important;
    position: absolute !important;
    white-space: nowrap !important;
    width: 1px !important;
}

.buttonTwo {
    border-radius: var(--border-button-small-border-radius, 4px) !important;
    font-size: var(--font-button-small-font-size, 14px) !important;
    line-height: var(--font-button-small-line-height, 22px) !important;
    letter-spacing: var(--font-button-small-letter-spacing, normal) !important;
    font-family: var(--font-button-small-font-family, Circular,-apple-system,BlinkMacSystemFont,Roboto,Helvetica Neue,sans-serif) !important;
    text-transform: var(--font-button-small-text-transform, undefined) !important;
    padding-top: var(--spacing-button-small-vertical, 6px) !important;
    padding-bottom: var(--spacing-button-small-vertical, 6px) !important;
    font-weight: var(--font-book-font-weight, 600) !important;
    border-width: var(--border-button-small-border-width, 1px) !important;
    padding-left: var(--spacing-button-small-horizontal, 15px) !important;
    padding-right: var(--spacing-button-small-horizontal, 15px) !important;
    box-shadow: var(--shadow-button-level0-box-shadow, none) !important;
    background: var(--color-buttons-inverse-color, #ffffff) !important;
    border-color: var(--color-buttons-inverse-border, transparent) !important;
    color: var(--color-buttons-inverse-text, #484848) !important;
    display: inline-block !important;
    margin: 0px !important;
    position: relative !important;
    text-align: center !important;
    text-decoration: none !important;
    -webkit-transition-property: background, border-color, color !important;
    -moz-transition-property: background, border-color, color !important;
    transition-property: background, border-color, color !important;
    -webkit-transition-duration: 0.2s !important;
    transition-duration: 0.2s !important;
    -webkit-transition-timing-function: ease-out !important;
    transition-timing-function: ease-out !important;
    width: auto !important;
    cursor: pointer !important;
    border-radius: 4px !important;
    font-size: 14px !important;
    line-height: 22px !important;
    letter-spacing: normal !important;
    font-family: Circular,-apple-system,BlinkMacSystemFont,Roboto,Helvetica Neue,sans-serif !important;
    text-transform: undefined !important;
    padding-top: 6px !important;
    padding-bottom: 6px !important;
    font-weight: 600 !important;
    border-width: 1px !important;
    border-style: solid !important;
    padding-left: 15px !important;
    padding-right: 15px !important;
    min-width: 48.54101966249685px !important;
    box-shadow: none !important;
    background: #ffffff !important;
    border-color: transparent !important;
    color: #484848 !important;
}

.buttonTwoSpan {
    font-size: var(--font-button-small-font-size, 14px) !important;
    line-height: var(--font-button-small-line-height, 22px) !important;
    letter-spacing: var(--font-button-small-letter-spacing, normal) !important;
    font-family: var(--font-button-small-font-family, Circular,-apple-system,BlinkMacSystemFont,Roboto,Helvetica Neue,sans-serif) !important;
    text-transform: var(--font-button-small-text-transform, undefined) !important;
    padding-top: var(--font-button-small-padding-top, undefined) !important;
    padding-bottom: var(--font-button-small-padding-bottom, undefined) !important;
    font-size: 14px !important;
    line-height: 22px !important;
    letter-spacing: normal !important;
    font-family: Circular,-apple-system,BlinkMacSystemFont,Roboto,Helvetica Neue,sans-serif !important;
    text-transform: undefined !important;
    color: inherit !important;
    padding-top: undefined !important;
    padding-bottom: undefined !important;
}

.buttonTwoText {
    display: -webkit-box !important;
    display: -moz-box !important;
    display: -ms-flexbox !important;
    display: -webkit-flex !important;
    display: flex !important;
}

.buttonTwoIcon {
    -ms-flex-item-align: center !important;
    display: -webkit-inline-box !important;
    display: -moz-inline-box !important;
    display: -ms-inline-flexbox !important;
    display: -webkit-inline-flex !important;
    display: inline-flex !important;
    -webkit-align-self: center !important;
    align-self: center !important;
}

.buttonTwoSave {
    -ms-flex-item-align: center !important;
    display: -webkit-inline-box !important;
    display: -moz-inline-box !important;
    display: -ms-inline-flexbox !important;
    display: -webkit-inline-flex !important;
    display: inline-flex !important;
    -webkit-align-self: center !important;
    align-self: center !important;
}

.buttonThreeContainer {
    box-shadow: 0 1px 1px 1px rgba(0, 0, 0, 0.14) !important;
    position: absolute !important;
    left: auto !important;
    right: 24px !important;
    top: 400px !important;
    -webkit-transform: translateY(0) !important;
    -ms-transform: translateY(0) !important;
    transform: translateY(0) !important;
    z-index: 1 !important;
}

.buttonThree {
    border-radius: var(--border-button-small-border-radius, 4px) !important;
    font-size: var(--font-button-small-font-size, 14px) !important;
    line-height: var(--font-button-small-line-height, 22px) !important;
    letter-spacing: var(--font-button-small-letter-spacing, normal) !important;
    font-family: var(--font-button-small-font-family, Circular,-apple-system,BlinkMacSystemFont,Roboto,Helvetica Neue,sans-serif) !important;
    text-transform: var(--font-button-small-text-transform, undefined) !important;
    padding-top: var(--spacing-button-small-vertical, 6px) !important;
    padding-bottom: var(--spacing-button-small-vertical, 6px) !important;
    font-weight: var(--font-book-font-weight, 600) !important;
    border-width: var(--border-button-small-border-width, 1px) !important;
    padding-left: var(--spacing-button-small-horizontal, 15px) !important;
    padding-right: var(--spacing-button-small-horizontal, 15px) !important;
    box-shadow: var(--shadow-button-level0-box-shadow, none) !important;
    background: var(--color-buttons-inverse-color, #ffffff) !important;
    border-color: var(--color-buttons-inverse-border, transparent) !important;
    color: var(--color-buttons-inverse-text, #484848) !important;
    display: inline-block !important;
    margin: 0px !important;
    position: relative !important;
    text-align: center !important;
    text-decoration: none !important;
    -webkit-transition-property: background, border-color, color !important;
    -moz-transition-property: background, border-color, color !important;
    transition-property: background, border-color, color !important;
    -webkit-transition-duration: 0.2s !important;
    transition-duration: 0.2s !important;
    -webkit-transition-timing-function: ease-out !important;
    transition-timing-function: ease-out !important;
    width: auto !important;
    cursor: pointer !important;
    border-radius: 4px !important;
    font-size: 14px !important;
    line-height: 22px !important;
    letter-spacing: normal !important;
    font-family: Circular,-apple-system,BlinkMacSystemFont,Roboto,Helvetica Neue,sans-serif !important;
    text-transform: undefined !important;
    padding-top: 6px !important;
    padding-bottom: 6px !important;
    font-weight: 600 !important;
    border-width: 1px !important;
    border-style: solid !important;
    padding-left: 15px !important;
    padding-right: 15px !important;
    min-width: 48.54101966249685px !important;
    box-shadow: none !important;
    background: #ffffff !important;
    border-color: transparent !important;
    color: #484848 !important;
}

.buttonThreeSpan {
    font-size: var(--font-button-small-font-size, 14px) !important;
    line-height: var(--font-button-small-line-height, 22px) !important;
    letter-spacing: var(--font-button-small-letter-spacing, normal) !important;
    font-family: var(--font-button-small-font-family, Circular,-apple-system,BlinkMacSystemFont,Roboto,Helvetica Neue,sans-serif) !important;
    text-transform: var(--font-button-small-text-transform, undefined) !important;
    padding-top: var(--font-button-small-padding-top, undefined) !important;
    padding-bottom: var(--font-button-small-padding-bottom, undefined) !important;
    font-size: 14px !important;
    line-height: 22px !important;
    letter-spacing: normal !important;
    font-family: Circular,-apple-system,BlinkMacSystemFont,Roboto,Helvetica Neue,sans-serif !important;
    text-transform: undefined !important;
    color: inherit !important;
    padding-top: undefined !important;
    padding-bottom: undefined !important;
}