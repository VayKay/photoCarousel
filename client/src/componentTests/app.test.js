/* eslint-disable no-undef */
import React from 'react';
import { shallow } from 'enzyme';
import App from '../components/app';

require('babel-core/register');
require('babel-polyfill');

describe('App', () => {
  it('should exist', () => {
    const component = shallow(<App />);

    expect(component.exists()).toBe(true);
  });

  it('should render correctly', async () => {
    const component = shallow(<App />);

    expect(component).toMatchSnapshot();
  });


  it('should update state of everything except profile on hover', () => {
    const component = shallow(<App />);
    component.setState({
      photoFiveOpacity:
      1,
      photoFourOpacity:
      1,
      photoThreeOpacity:
      1,
      photoTwoOpacity:
      1,
      profileOpacity:
      1,
    });
    component.instance().hoverHandler('profile');
    expect(component.state().photoTwoOpacity).toEqual(0.7);
    expect(component.state().photoThreeOpacity).toEqual(0.7);
    expect(component.state().photoFourOpacity).toEqual(0.7);
    expect(component.state().photoFiveOpacity).toEqual(0.7);
  });

  it('should update state of everything except pictureTwo on hover', () => {
    const component = shallow(<App />);
    component.setState({
      photoFiveOpacity:
      1,
      photoFourOpacity:
      1,
      photoThreeOpacity:
      1,
      photoTwoOpacity:
      1,
      profileOpacity:
      1,
    });
    component.instance().hoverHandler('two');
    expect(component.state().profileOpacity).toEqual(0.7);
    expect(component.state().photoThreeOpacity).toEqual(0.7);
    expect(component.state().photoFourOpacity).toEqual(0.7);
    expect(component.state().photoFiveOpacity).toEqual(0.7);
  });

  it('should update state of everything except pictureThree on hover', () => {
    const component = shallow(<App />);
    component.setState({
      photoFiveOpacity:
      1,
      photoFourOpacity:
      1,
      photoThreeOpacity:
      1,
      photoTwoOpacity:
      1,
      profileOpacity:
      1,
    });
    component.instance().hoverHandler('three');
    expect(component.state().profileOpacity).toEqual(0.7);
    expect(component.state().photoTwoOpacity).toEqual(0.7);
    expect(component.state().photoFourOpacity).toEqual(0.7);
    expect(component.state().photoFiveOpacity).toEqual(0.7);
  });

  it('should update state of everything except pictureFour on hover', () => {
    const component = shallow(<App />);
    component.setState({
      photoFiveOpacity:
      1,
      photoFourOpacity:
      1,
      photoThreeOpacity:
      1,
      photoTwoOpacity:
      1,
      profileOpacity:
      1,
    });
    component.instance().hoverHandler('four');
    expect(component.state().profileOpacity).toEqual(0.7);
    expect(component.state().photoTwoOpacity).toEqual(0.7);
    expect(component.state().photoThreeOpacity).toEqual(0.7);
    expect(component.state().photoFiveOpacity).toEqual(0.7);
  });

  it('should update state of everything except pictureFive on hover', () => {
    const component = shallow(<App />);
    component.setState({
      photoFiveOpacity:
      1,
      photoFourOpacity:
      1,
      photoThreeOpacity:
      1,
      photoTwoOpacity:
      1,
      profileOpacity:
      1,
    });
    component.instance().hoverHandler('five');
    expect(component.state().profileOpacity).toEqual(0.7);
    expect(component.state().photoTwoOpacity).toEqual(0.7);
    expect(component.state().photoThreeOpacity).toEqual(0.7);
    expect(component.state().photoFourOpacity).toEqual(0.7);
  });

  it('should update state of everything when mouse is not hovering on any picture', () => {
    const component = shallow(<App />);
    component.setState({
      photoFiveOpacity:
      1,
      photoFourOpacity:
      0.7,
      photoThreeOpacity:
      0.7,
      photoTwoOpacity:
      0.7,
      profileOpacity:
      0.7,
    });

    component.instance().hoverHandler('clear');
    expect(component.state().profileOpacity).toEqual(1);
    expect(component.state().photoTwoOpacity).toEqual(1);
    expect(component.state().photoThreeOpacity).toEqual(1);
    expect(component.state().photoFourOpacity).toEqual(1);
  });

  it('should update state on click', () => {
    const component = shallow(<App />);
    component.setState({
      modalFocus: 3,
      modalView: false,
    });
    component.instance().clickHandler('profile');

    expect(component.state().modalFocus).toEqual(0);
    expect(component.state().modalView).toEqual(true);
  });

  it('renders all children components except modalView when there are atleast than 5 pictures', () => {
    const component = shallow(<App />);
    component.setState({
      photos: [{
        caption:
        'Et esse laudantium sunt et.',
        createdAt:
        '2019-07-10T01:42:08.000Z',
        id:
        5,
        listing_id:
        1,
        photoUrl:
        'http://lorempixel.com/1440/960/city',
        priority:
        0,
        updatedAt:
        '2019-07-10T01:42:08.000Z',
      },
      {
        caption:
        'Et esse laudantium sunt et.',
        createdAt:
        '2019-07-10T01:42:08.000Z',
        id:
        5,
        listing_id:
        1,
        photoUrl:
        'http://lorempixel.com/1440/960/city',
        priority:
        1,
        updatedAt:
        '2019-07-10T01:42:08.000Z',
      },
      {
        caption:
        'Et esse laudantium sunt et.',
        createdAt:
        '2019-07-10T01:42:08.000Z',
        id:
        5,
        listing_id:
        1,
        photoUrl:
        'http://lorempixel.com/1440/960/city',
        priority:
        2,
        updatedAt:
        '2019-07-10T01:42:08.000Z',
      },
      {
        caption:
        'Et esse laudantium sunt et.',
        createdAt:
        '2019-07-10T01:42:08.000Z',
        id:
        5,
        listing_id:
        1,
        photoUrl:
        'http://lorempixel.com/1440/960/city',
        priority:
        3,
        updatedAt:
        '2019-07-10T01:42:08.000Z',
      },
      {
        caption:
        'Et esse laudantium sunt et.',
        createdAt:
        '2019-07-10T01:42:08.000Z',
        id:
        5,
        listing_id:
        1,
        photoUrl:
        'http://lorempixel.com/1440/960/city',
        priority:
        4,
        updatedAt:
        '2019-07-10T01:42:08.000Z',
      },
      ],
      currWidth: 10000,
    });
    expect(component.find('ProfilePicture').length).toEqual(1);
    expect(component.find('SubPictures').length).toEqual(1);
    expect(component.find('ExtraPictures').length).toEqual(1);
    expect(component.find('Slideshow').length).toEqual(0);
  });

  it('only render Profile and SubPicture components when there are less than 5 pictures but atleast than 3', () => {
    const component = shallow(<App />);
    component.setState({
      photos: [{
        caption:
        'Et esse laudantium sunt et.',
        createdAt:
        '2019-07-10T01:42:08.000Z',
        id:
        5,
        listing_id:
        1,
        photoUrl:
        'http://lorempixel.com/1440/960/city',
        priority:
        0,
        updatedAt:
        '2019-07-10T01:42:08.000Z',
      },
      {
        caption:
        'Et esse laudantium sunt et.',
        createdAt:
        '2019-07-10T01:42:08.000Z',
        id:
        5,
        listing_id:
        1,
        photoUrl:
        'http://lorempixel.com/1440/960/city',
        priority:
        1,
        updatedAt:
        '2019-07-10T01:42:08.000Z',
      },
      {
        caption:
        'Et esse laudantium sunt et.',
        createdAt:
        '2019-07-10T01:42:08.000Z',
        id:
        5,
        listing_id:
        1,
        photoUrl:
        'http://lorempixel.com/1440/960/city',
        priority:
        2,
        updatedAt:
        '2019-07-10T01:42:08.000Z',
      },
      ],
    });

    expect(component.find('ProfilePicture').length).toEqual(1);
    expect(component.find('SubPictures').length).toEqual(1);
    expect(component.find('ExtraPictures').length).toEqual(0);
    expect(component.find('Slideshow').length).toEqual(0);
  });

  it('only render ProfilePicture when there is atleast only 1 picture', () => {
    const component = shallow(<App />);
    component.setState({
      photos: [{
        caption:
        'Et esse laudantium sunt et.',
        createdAt:
        '2019-07-10T01:42:08.000Z',
        id:
        5,
        listing_id:
        1,
        photoUrl:
        'http://lorempixel.com/1440/960/city',
        priority:
        0,
        updatedAt:
        '2019-07-10T01:42:08.000Z',
      },
      ],
    });

    expect(component.find('ProfilePicture').length).toEqual(1);
    expect(component.find('SubPictures').length).toEqual(0);
    expect(component.find('ExtraPictures').length).toEqual(0);
    expect(component.find('Slideshow').length).toEqual(0);
  });

  it('Should not render ExtraPictures when window is below first threshold', () => {
    const component = shallow(<App />);
    component.setState({
      photos: [{
        caption:
        'Et esse laudantium sunt et.',
        createdAt:
        '2019-07-10T01:42:08.000Z',
        id:
        5,
        listing_id:
        1,
        photoUrl:
        'http://lorempixel.com/1440/960/city',
        priority:
        0,
        updatedAt:
        '2019-07-10T01:42:08.000Z',
      },
      {
        caption:
        'Et esse laudantium sunt et.',
        createdAt:
        '2019-07-10T01:42:08.000Z',
        id:
        5,
        listing_id:
        1,
        photoUrl:
        'http://lorempixel.com/1440/960/city',
        priority:
        1,
        updatedAt:
        '2019-07-10T01:42:08.000Z',
      },
      {
        caption:
        'Et esse laudantium sunt et.',
        createdAt:
        '2019-07-10T01:42:08.000Z',
        id:
        5,
        listing_id:
        1,
        photoUrl:
        'http://lorempixel.com/1440/960/city',
        priority:
        2,
        updatedAt:
        '2019-07-10T01:42:08.000Z',
      },
      {
        caption:
        'Et esse laudantium sunt et.',
        createdAt:
        '2019-07-10T01:42:08.000Z',
        id:
        5,
        listing_id:
        1,
        photoUrl:
        'http://lorempixel.com/1440/960/city',
        priority:
        3,
        updatedAt:
        '2019-07-10T01:42:08.000Z',
      },
      {
        caption:
        'Et esse laudantium sunt et.',
        createdAt:
        '2019-07-10T01:42:08.000Z',
        id:
        5,
        listing_id:
        1,
        photoUrl:
        'http://lorempixel.com/1440/960/city',
        priority:
        4,
        updatedAt:
        '2019-07-10T01:42:08.000Z',
      },
      ],
      currWidth: component.state().threshold - 1,
    });

    expect(component.find('ProfilePicture').length).toEqual(1);
    expect(component.find('SubPictures').length).toEqual(1);
    expect(component.find('ExtraPictures').length).toEqual(0);
    expect(component.find('Slideshow').length).toEqual(0);
  });

  it('Should not render SubPictures when window is below last threshold', () => {
    const component = shallow(<App />);
    component.setState({
      photos: [{
        caption:
        'Et esse laudantium sunt et.',
        createdAt:
        '2019-07-10T01:42:08.000Z',
        id:
        5,
        listing_id:
        1,
        photoUrl:
        'http://lorempixel.com/1440/960/city',
        priority:
        0,
        updatedAt:
        '2019-07-10T01:42:08.000Z',
      },
      {
        caption:
        'Et esse laudantium sunt et.',
        createdAt:
        '2019-07-10T01:42:08.000Z',
        id:
        5,
        listing_id:
        1,
        photoUrl:
        'http://lorempixel.com/1440/960/city',
        priority:
        1,
        updatedAt:
        '2019-07-10T01:42:08.000Z',
      },
      {
        caption:
        'Et esse laudantium sunt et.',
        createdAt:
        '2019-07-10T01:42:08.000Z',
        id:
        5,
        listing_id:
        1,
        photoUrl:
        'http://lorempixel.com/1440/960/city',
        priority:
        2,
        updatedAt:
        '2019-07-10T01:42:08.000Z',
      },
      {
        caption:
        'Et esse laudantium sunt et.',
        createdAt:
        '2019-07-10T01:42:08.000Z',
        id:
        5,
        listing_id:
        1,
        photoUrl:
        'http://lorempixel.com/1440/960/city',
        priority:
        3,
        updatedAt:
        '2019-07-10T01:42:08.000Z',
      },
      {
        caption:
        'Et esse laudantium sunt et.',
        createdAt:
        '2019-07-10T01:42:08.000Z',
        id:
        5,
        listing_id:
        1,
        photoUrl:
        'http://lorempixel.com/1440/960/city',
        priority:
        4,
        updatedAt:
        '2019-07-10T01:42:08.000Z',
      },
      ],
      currWidth: component.state().lastThreshold - 1,
    });

    expect(component.find('ProfilePicture').length).toEqual(1);
    expect(component.find('SubPictures').length).toEqual(0);
    expect(component.find('ExtraPictures').length).toEqual(0);
    expect(component.find('Slideshow').length).toEqual(0);
  });

  it('Should render Slideshow component and change focus on component click', () => {
    const component = shallow(<App />);
    component.setState({
      photos: [{
        caption:
        'Et esse laudantium sunt et.',
        createdAt:
        '2019-07-10T01:42:08.000Z',
        id:
        5,
        listing_id:
        1,
        photoUrl:
        'http://lorempixel.com/1440/960/city',
        priority:
        0,
        updatedAt:
        '2019-07-10T01:42:08.000Z',
      },
      {
        caption:
        'Et esse laudantium sunt et.',
        createdAt:
        '2019-07-10T01:42:08.000Z',
        id:
        5,
        listing_id:
        1,
        photoUrl:
        'http://lorempixel.com/1440/960/city',
        priority:
        1,
        updatedAt:
        '2019-07-10T01:42:08.000Z',
      },
      {
        caption:
        'Et esse laudantium sunt et.',
        createdAt:
        '2019-07-10T01:42:08.000Z',
        id:
        5,
        listing_id:
        1,
        photoUrl:
        'http://lorempixel.com/1440/960/city',
        priority:
        2,
        updatedAt:
        '2019-07-10T01:42:08.000Z',
      },
      {
        caption:
        'Et esse laudantium sunt et.',
        createdAt:
        '2019-07-10T01:42:08.000Z',
        id:
        5,
        listing_id:
        1,
        photoUrl:
        'http://lorempixel.com/1440/960/city',
        priority:
        3,
        updatedAt:
        '2019-07-10T01:42:08.000Z',
      },
      {
        caption:
        'Et esse laudantium sunt et.',
        createdAt:
        '2019-07-10T01:42:08.000Z',
        id:
        5,
        listing_id:
        1,
        photoUrl:
        'http://lorempixel.com/1440/960/city',
        priority:
        4,
        updatedAt:
        '2019-07-10T01:42:08.000Z',
      },
      ],
    });
    component.instance().clickHandler('profile');

    expect(component.find('Slideshow').length).toEqual(1);
    expect(component.state().modalFocus).toEqual(0);
  });

  it('Should exit modalSlideshow when you click the exit button', () => {
    const component = shallow(<App />);
    component.setState({
      modalView: true,
      modalFocus: 0,
    });

    component.instance().exitModal();

    expect(component.find('Slideshow').length).toEqual(0);
  });
});
