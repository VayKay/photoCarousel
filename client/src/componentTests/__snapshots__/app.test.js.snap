// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
<div
  className="photoCarousel"
>
  <div
    className="buttonThreeContainer"
  >
    <button
      aria-busy="false"
      className="buttonThree"
      data-veloute="hero-view-photos-button"
      onClick={[Function]}
      type="button"
    >
      <span
        className="buttonThreeSpan"
      >
        View Photos
      </span>
    </button>
  </div>
  <div
    className="topButtonsContainer"
  >
    <div
      className="topButtonsInner"
    >
      <div
        className="buttonOneContainer"
      >
        <div
          className="buttonOneInner"
        >
          <button
            aria-busy="false"
            className="buttonOne"
            type="button"
          >
            <span
              className="buttonOneSpan"
            >
              <div
                className="buttonOneText"
                style={
                  Object {
                    "justifyContent": "center",
                  }
                }
              >
                <div
                  className="buttonOneIcon"
                >
                  <div
                    style={
                      Object {
                        "marginRight": "12px",
                      }
                    }
                  >
                    <svg
                      aria-hidden="true"
                      fill="currentColor"
                      fillOpacity="0"
                      focusable="false"
                      role="presentation"
                      stroke="currentColor"
                      strokeLinecap="round"
                      strokeLinejoin="round"
                      strokeWidth="2.25"
                      style={
                        Object {
                          "display": "block",
                          "height": "15px",
                          "overflow": "visible",
                          "width": "15px",
                        }
                      }
                      viewBox="0 0 24 24"
                    >
                      <g
                        fillRule="evenodd"
                      >
                        <path
                          d="m11.5 16v-15"
                        />
                        <path
                          d="m7.5 4 4-3 4 3"
                        />
                        <path
                          d="m5.4 9.5h-3.9v14h20v-14h-3.1"
                        />
                      </g>
                    </svg>
                  </div>
                </div>
                <div
                  className="buttonOneShare"
                >
                  <div>
                    Share
                  </div>
                </div>
              </div>
            </span>
          </button>
        </div>
      </div>
      <div
        className="buttonTwoContainer"
        style={
          Object {
            "marginLeft": "18px",
          }
        }
      >
        <span>
          <div
            className="buttonTwoInner"
          >
            <span
              aria-hidden="true"
              className="buttonTwoAccessSpan"
              id="save-button-accessible-text"
            >
              Save this listing.
            </span>
            <button
              aria-busy="false"
              aria-describedby="save-button-accessible-text"
              className="buttonTwo"
              type="button"
            >
              <span
                className="buttonTwoSpan"
              >
                <div
                  className="buttonTwoText"
                  style={
                    Object {
                      "justifyContent": "center",
                    }
                  }
                >
                  <div
                    className="buttonTwoIcon"
                  >
                    <div
                      style={
                        Object {
                          "marginRight": "12px",
                        }
                      }
                    >
                      <svg
                        aria-label="Save this listing."
                        fill="currentColor"
                        fillOpacity="0"
                        focusable="false"
                        role="img"
                        stroke="#484848"
                        strokeLinecap="round"
                        strokeLinejoin="round"
                        strokeWidth="2.25"
                        style={
                          Object {
                            "display": "block",
                            "height": "15px",
                            "overflow": "visible",
                            "width": "15px",
                          }
                        }
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="m17.5 2.9c-2.1 0-4.1 1.3-5.4 2.8-1.6-1.6-3.8-3.2-6.2-2.7-1.5.2-2.9 1.2-3.6 2.6-2.3 4.1 1 8.3 3.9 11.1 1.4 1.3 2.8 2.5 4.3 3.6.4.3 1.1.9 1.6.9s1.2-.6 1.6-.9c3.2-2.3 6.6-5.1 8.2-8.8 1.5-3.4 0-8.6-4.4-8.6"
                          strokeLinejoin="round"
                        />
                      </svg>
                    </div>
                  </div>
                  <div
                    className="buttonTwoSave"
                  >
                    <div>
                      Save
                    </div>
                  </div>
                </div>
              </span>
            </button>
          </div>
        </span>
      </div>
    </div>
  </div>
</div>
`;
